apps/console/src/components/pages/protected/controls/table/controls-table.tsx

```diff
@@ -87,7 +87,7 @@ const ControlsTable: React.FC<TControlsTableProps> = ({ active, setActive }) =>
       }
 
       if (key === 'hasProgramsWith') {
-        return { hasProgramsWith: [{ id: value as string }] } as ControlWhereInput
+        return { hasProgramsWith: (value as string[]).map((v) => ({ id: v })) } as ControlWhereInput
```

Comment:
This isn't quite right, this makes an `and` between the ids, so you get no results back

```
"hasProgramsWith": [
          {
            "id": "01K8QFMM8ZK90ZV41WN5DCGST1"
          },
          {
            "id": "01K8QXQK04GGVY5A9Y562N5PKF"
          }
        ]
 ```
. Instead it should take the string array and use `idIn`, something like:
```suggestion
        return { hasProgramsWith: (({ idIn: value })) } as ControlWhereInput
```

This is true for all the updates to `hasProgramsWith`

Reply:
changed for all